src/main/java/Matchers/TagMatcher.java

```diff
@@ -1,10 +1,44 @@
 package main.java.Matchers;
 
+import main.java.EntityInterfaces.RecipeImpl;
+import main.java.Entities.Tag;
+import main.java.Storages.RecipeStorage;
+import main.java.Entities.Recipe;
+import main.java.Entities.Item;
+
+
+
+import java.util.List;
+
 /**
- * Match recipes with given set of tags
+ * Match recipes based on tags
  */
-public class TagMatcher {
+public class TagMatcher implements MatcherImpl {
+    List<Tag> tags;
+
+    public TagMatcher(List<Tag> tags) {
+        this.tags = tags;
+    }
+
+    @Override
+    /**
+     * Checks if item/ingredient contains any of the tags in the matcher
+     */
+    public Boolean matches(Recipe recipe) {
+        for (Tag tag: this.tags) {
```

Comment:
Presumably the logic here should be that the recipe has all of the given tags. The checking for loop should look something like:

```
for tag in tags:
  matchedTag = false
  for ingredient in items:
    if ingredient has tag:
      matchedTag = true
      break;

  if !matchedTag:
    return false
    
return true
```